applications/GeoMechanicsApplication/custom_strategies/schemes/newmark_quasistatic_T_scheme.hpp

```diff
@@ -0,0 +1,154 @@
+    int Check(const ModelPart& rModelPart) const override
+    {
+        KRATOS_TRY
+
+        BaseType::Check(rModelPart);
+
+        //check that variables are correctly allocated
+        for (const auto& rNode : rModelPart.Nodes())
+        {
+            if (rNode.SolutionStepsDataHas(TEMPERATURE) == false)
+                KRATOS_ERROR << "TEMPERATURE variable is not allocated for node "
+                             << rNode.Id()
+                             << std::endl;
+
+            if (rNode.SolutionStepsDataHas(DT_TEMPERATURE) == false)
+                KRATOS_ERROR << "DT_TEMPERATURE variable is not allocated for node "
+                             << rNode.Id()
+                             << std::endl;
+
+            if (rNode.HasDofFor(TEMPERATURE) == false)
+                KRATOS_ERROR << "missing TEMPERATURE dof on node "
+                             << rNode.Id()
+                             << std::endl;
+        }
+
+        //check for minimum value of the buffer index.
+        if (rModelPart.GetBufferSize() < 2)
+            KRATOS_ERROR << "insufficient buffer size. Buffer size should be greater than 2. Current size is "
+                         << rModelPart.GetBufferSize()
+                         << std::endl;
+
+        // Check beta, gamma and theta
+        if (mBeta <= 0.0 || mGamma<= 0.0 || mTheta <= 0.0)
+            KRATOS_ERROR << "Some of the scheme variables: beta, gamma or theta has an invalid value "
+                         << std::endl;
+
+        return 0;
+
+        KRATOS_CATCH( "" )
+    }
+
+
+    // ============================================================================================
+    // ============================================================================================
+    void FinalizeSolutionStep(
+        ModelPart& rModelPart,
+        TSystemMatrixType& A,
+        TSystemVectorType& Dx,
+        TSystemVectorType& b) override
+    {
+        KRATOS_TRY
+
+        MotherType::FinalizeSolutionStepActiveEntities(rModelPart,A,Dx,b);
+
+        KRATOS_CATCH("")
+    }
+
+
+protected:
+
+    /// Member Variables
+    // ============================================================================================
+    // ============================================================================================
+    inline void UpdateVariablesDerivatives(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        //Update DtPressure
+        block_for_each(rModelPart.Nodes(), [&](Node& rNode){
+            const double DeltaTemperature =  rNode.FastGetSolutionStepValue(TEMPERATURE)
+                                        - rNode.FastGetSolutionStepValue(TEMPERATURE, 1);
+            const auto &PreviousDtTemperature = rNode.FastGetSolutionStepValue(DT_TEMPERATURE, 1);
+
+            rNode.FastGetSolutionStepValue(DT_TEMPERATURE) =  1.0/(mTheta*mDeltaTime)*(DeltaTemperature - (1.0-mTheta)*mDeltaTime*PreviousDtTemperature);
```

Comment:
dimensional analysis raises doubt about this formula for mTheta other than 1.

Reply:
This problem is overal in the code, also for pressure and deformation.
I think it is not suitable to fix it in the thermal branch. I will make a new issue and address this issue separately (in a new branch).